mne/stats/regression.py

```diff
@@ -392,3 +392,254 @@ def _make_evokeds(coefs, conds, cond_length, tmin_s, tmax_s, info):
+        """Fit a STRF model.
+
+        Fit a receptive field using time lags and a custom estimator or
+        pipeline. This implementation uses Ridge regression and scikit-learn.
+        It creates time lags for the input matrix, then does cross validation
+        to fit a STRF model.
+
+        Parameters
+        ----------
+        delays : array, shape (n_delays,)
+            The delays to include when creating time lags. The input array X
+            will end up having shape (n_feats * n_delays, n_times)
+        est : model estimator | pipeline with estimator
+            The estimator to use for fitting. This is any object that contains
+            a `fit` and `predict` method, which takes inputs of the form
+            (X, y), and which creates a `.coef_` attribute upon fitting.
+            This may be a pipeline, in which case the final estimator must
+            follow the same conventions described above.
+
+        References
+        ----------
+        .. [1] Smith, N. J., & Kutas, M. (2015). Regression-based estimation of
+               ERP waveforms: II. Non-linear effects, overlap correction, and
+               practical considerations. Psychophysiology, 52(2), 169-189.
+        .. [2] Theunissen, F. E. et al. Estimating spatio-temporal receptive
+               fields of auditory and visual neurons from their responses to
+               natural stimuli. Network 12, 289-316 (2001).
+        .. [3] Willmore, B. & Smyth, D. Methods for first-order kernel
+               estimation: simple-cell receptive fields from responses to
+               natural scenes. Network 14, 553-77 (2003).
+        """
+        self.delays = np.array([0]) if delays is None else delays
+        self.n_delays = len(self.delays)
+        self.est = _check_estimator(est)
+        self.preproc_y = preproc_y
+
+    def fit(self, raw, continuous=None, events=None, event_id=None,
+            picks=None, fit_ixs=None, continuous_names=None, verbose=False):
+        """Fit the model.
+
+        Fits a receptive field model. Model results are stored as attributes.
+
+        Parameters
+        ----------
+        raw : instance of MNE Raw
+            The data on which we want to fit a regression model.
+        events : array, shape (n_events, 3)
+            An MNE events array specifying indices for event onsets
+        event_id : dictionary | None
+            A dictionary of (event_name: event_int) pairs, corresponding to the
+            mapping from event name strings to the 3rd column of events.
+        continuous : array, shape (n_feats, n_times)
+            Continuous input features in the regression.
+        continuous_names : array, shape (n_feats,) | None
+            Names for the input continuous variables.
+        picks : array, shape (n_picks,) | None
+            Indices for channels to use in model fitting. If None, all channels
+            will be fit.
+        fit_ixs : array, dtype int | None
```

Comment:
Here I am imagining it being used for cross-validation as opposed to decimation. That way it would work with the `train, test` outputs of a cross-validation iterator. Does that make sense?